### ISSUE TITLE
Feature Request: alternative storage adapters

### ISSUE THREAD

**Issue by mikestaub** [Cameri/nostream#147]
have you considered using another SQL engine like cockroach db?

I worry that postgres may have trouble scaling to millions of users without significant devops effort

https://github.com/cockroachdb/cockroach/blob/master/docs/design.md

**Comment by Cameri** [comment]
@mikestaub yes, I've considered it and like the idea. I was actually thinking about this recently
